### DISCUSSION TITLE
PowerInfer: Built on llama.cpp, Now 11x Faster

### DISCUSSION THREAD

**Discussion by bobqianic** [ggml-org/llama.cpp#4548] [General]
I think the main breakthrough is that it can arrange the position of weight parameters more scientifically based on the frequency of neuron activation, _**placing the frequently activated weights in faster-reading caches to improve inference speed**_. They developed a **_Neuron-aware Operator_** that can bypass neurons that are not activated, and also developed an offline profiling technique that allows the _**Neuron Placement Policy**_ to be predetermined.

Github: https://github.com/SJTU-IPADS/PowerInfer

Paper: https://ipads.se.sjtu.edu.cn/_media/publications/powerinfer-20231219.pdf

Relevant discussions: #4543 #4542 #4534

**Comment by joseph777111** [comment]
@ggerganov,what are your thoughts on this? Could this be useful for llama.cpp? 🤔

**Reply by bobqianic** [reply]
See https://github.com/ggerganov/llama.cpp/discussions/4534#discussioncomment-7920878

**Reply by joseph777111** [reply]
Thanks 🤘😁
